### PR TITLE
Check web_contents for nullptr

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Fixes https://github.com/brave/brave-browser/issues/7861

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues/7861) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
Please follow steps at https://github.com/brave/brave-browser/issues/7861 

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by AlexeyBarabash** [comment]
Here is what happened.
1. 
```
void BraveShieldsWebContentsObserver::DispatchBlockedEvent(...) {
   // render_process_id=-1
...
   WebContents* web_contents = GetWebContents(...)
   // web_contents=0x0
   DispatchBlockedEventForWebContents(block_type, subresource, web_contents);
...
}
```

2.
```
// brave/components/brave_shields/browser/brave_shields_web_contents_observer_android.cc
void BraveShieldsWebContentsObserver::DispatchBlockedEventForWebContents(...) {
...
   // web_contents=0x0
   TabAndroid* tab = TabAndroid::FromWebContents(web_contents);
...
}
``` 

3. 
```
// src/chrome/browser/android/tab_android.cc
TabAndroid* TabAndroid::FromWebContents() {
   // web_contents=0x0
}
```
4. 
```
// same file
static TabAndroid* FromWebContents(const WebContents* contents) {
    // contents=0x0
    TabAndroidHelper* helper =
        static_cast<TabAndroidHelper*>(contents->GetUserData(UserDataKey()));
}
```
null pointer access.

From the other side, desktop implementation has the guard
```
// brave/components/brave_shields/browser/brave_shields_web_contents_observer.cc
void BraveShieldsWebContentsObserver::DispatchBlockedEventForWebContents(...) {
  if (!web_contents) {
    return;
  }
}
```

After this change it was possible to see webview crash when scroll to the bottom of the page on debug build.
That webview crash can be disabled by https://github.com/brave/brave-core/pull/4389 .

File: components/brave_shields/browser/brave_shields_web_contents_observer_android.cc [hunk 1]
```diff
@@ -19,6 +19,9 @@ namespace brave_shields {
 void BraveShieldsWebContentsObserver::DispatchBlockedEventForWebContents(
     const std::string& block_type, const std::string& subresource,
     WebContents* web_contents) {
+  if (!web_contents) {
+    return;
+  }
```
Comment (yrliou) [inline components/brave_shields/browser/brave_shields_web_contents_observer_android.cc]: super nit: could you add a newline after this, thanks.

Comment (yrliou) [reply components/brave_shields/browser/brave_shields_web_contents_observer_android.cc]: BTW, you could skip CI when addressing this comment, thanks.

Comment (AlexeyBarabash) [reply components/brave_shields/browser/brave_shields_web_contents_observer_android.cc]: @yrliou thanks, addressed in https://github.com/brave/brave-core/compare/5aa2ac7326caa1855b1841d7c3a045f0e7c2efb3..f6c6202eec8d354bd90c246f9f6d3a6dd0f4a39b

**Comment by AlexeyBarabash** [comment]
I have re-run CI because commit was marked with red cross after force-push.

**Pull request merged**